webapp/channels/src/packages/mattermost-redux/src/actions/emojis.ts

```diff
@@ -69,16 +72,46 @@ export function getCustomEmojiByName(name: string): ActionFunc {
 }
 
 export function getCustomEmojisByName(names: string[]): ActionFunc {
-    return async (dispatch: DispatchFunc) => {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
         if (!names || names.length === 0) {
             return {data: true};
         }
 
-        const promises: Array<Promise<ActionResult|ActionResult[]>> = [];
-        names.forEach((name) => promises.push(dispatch(getCustomEmojiByName(name))));
+        let data;
+        try {
+            data = await Client4.getCustomEmojisByNames(names);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
 
-        await Promise.all(promises);
-        return {data: true};
+        const actions: AnyAction[] = [{
+            type: EmojiTypes.RECEIVED_CUSTOM_EMOJIS,
+            data,
+        }];
+
+        if (data.length !== names.length) {
+            for (const name of names) {
```

Comment:
I was considering making this loop more efficient, but it seems unnecessary given we shouldn't run into this case often with how we use this API

Reply:
0/5 about making this more efficient. I feel with sets will get a bit more readable, but I also get your point this will get called not often.

Reply:
Good point. I'll see if I can make that a bit easier to read